common/cpp/react/renderer/components/rnscreens/RNSScreenStackHeaderConfigShadowNode.cpp

```diff
@@ -5,6 +5,18 @@ namespace facebook::react {
 extern const char RNSScreenStackHeaderConfigComponentName[] =
     "RNSScreenStackHeaderConfig";
 
+void RNSScreenStackHeaderConfigShadowNode::layout(LayoutContext layoutContext) {
+  YogaLayoutableShadowNode::layout(layoutContext);
+  applyFrameCorrections();
+}
+
+void RNSScreenStackHeaderConfigShadowNode::applyFrameCorrections() {
+  ensureUnsealed();
+
+  const auto &stateData = getStateData();
+  layoutMetrics_.frame.origin.y = -stateData.frameSize.height;
```

Comment:
Here we just simply offset the whole headerconfig frame by its height. This code works under assumption that header will be placed at `(0, 0)` of its `Screen` parent *initially*. Maybe a comment on this assumption would be helpful here.